{
    "dataType": "CVE_RECORD",
    "dataVersion": "5.0",
    "cveMetadata": {
        "cveId": "CVE-2023-31136",
        "assignerOrgId": "a0819718-46f1-4df5-94e2-005712e83aaa",
        "state": "PUBLISHED",
        "assignerShortName": "GitHub_M",
        "dateReserved": "2023-04-24T21:44:10.417Z",
        "datePublished": "2023-05-09T13:37:38.193Z",
        "dateUpdated": "2023-05-09T13:37:38.193Z"
    },
    "containers": {
        "cna": {
            "title": "PostgresNIO processes unencrypted bytes from man-in-the-middle",
            "problemTypes": [
                {
                    "descriptions": [
                        {
                            "cweId": "CWE-522",
                            "lang": "en",
                            "description": "CWE-522: Insufficiently Protected Credentials",
                            "type": "CWE"
                        }
                    ]
                }
            ],
            "metrics": [
                {
                    "cvssV3_1": {
                        "attackComplexity": "HIGH",
                        "attackVector": "NETWORK",
                        "availabilityImpact": "NONE",
                        "baseScore": 3.7,
                        "baseSeverity": "LOW",
                        "confidentialityImpact": "LOW",
                        "integrityImpact": "NONE",
                        "privilegesRequired": "NONE",
                        "scope": "UNCHANGED",
                        "userInteraction": "NONE",
                        "vectorString": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:L/I:N/A:N",
                        "version": "3.1"
                    }
                }
            ],
            "references": [
                {
                    "name": "https://github.com/vapor/postgres-nio/security/advisories/GHSA-9cfh-vx93-84vv",
                    "tags": [
                        "x_refsource_CONFIRM"
                    ],
                    "url": "https://github.com/vapor/postgres-nio/security/advisories/GHSA-9cfh-vx93-84vv"
                },
                {
                    "name": "https://github.com/apple/swift-nio/pull/2419",
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://github.com/apple/swift-nio/pull/2419"
                },
                {
                    "name": "https://github.com/vapor/postgres-nio/commit/2df54bc94607f44584ae6ffa74e3cd754fffafc7",
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://github.com/vapor/postgres-nio/commit/2df54bc94607f44584ae6ffa74e3cd754fffafc7"
                },
                {
                    "name": "https://github.com/advisories/GHSA-467w-rrqc-395f",
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://github.com/advisories/GHSA-467w-rrqc-395f"
                },
                {
                    "name": "https://github.com/advisories/GHSA-735f-7qx4-jqq5",
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://github.com/advisories/GHSA-735f-7qx4-jqq5"
                },
                {
                    "name": "https://github.com/vapor/postgres-nio/releases/tag/1.14.2",
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://github.com/vapor/postgres-nio/releases/tag/1.14.2"
                },
                {
                    "name": "https://www.postgresql.org/support/security/CVE-2021-23214/",
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://www.postgresql.org/support/security/CVE-2021-23214/"
                },
                {
                    "name": "https://www.postgresql.org/support/security/CVE-2021-23222/",
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://www.postgresql.org/support/security/CVE-2021-23222/"
                }
            ],
            "affected": [
                {
                    "vendor": "vapor",
                    "product": "postgres-nio",
                    "versions": [
                        {
                            "version": "< 1.14.2",
                            "status": "affected"
                        }
                    ]
                }
            ],
            "providerMetadata": {
                "orgId": "a0819718-46f1-4df5-94e2-005712e83aaa",
                "shortName": "GitHub_M",
                "dateUpdated": "2023-05-09T13:37:38.193Z"
            },
            "descriptions": [
                {
                    "lang": "en",
                    "value": "PostgresNIO is a Swift client for PostgreSQL. Any user of PostgresNIO prior to version 1.14.2 connecting to servers with TLS enabled is vulnerable to a man-in-the-middle attacker injecting false responses to the client's first few queries, despite the use of TLS certificate verification and encryption. The vulnerability is addressed in PostgresNIO versions starting from 1.14.2. There are no known workarounds for unpatched users."
                }
            ],
            "source": {
                "advisory": "GHSA-9cfh-vx93-84vv",
                "discovery": "UNKNOWN"
            }
        }
    }
}